lib/signers/v4.js

```diff
@@ -63,6 +63,9 @@ AWS.Signers.V4 = inherit(AWS.Signers.RequestSigner, {
     if (this.request.headers['Content-MD5']) {
       qs['Content-MD5'] = this.request.headers['Content-MD5'];
     }
+    if (this.request.headers['Cache-Control']) {
```

Comment:
So looking at the commit log, it looks like this is the 2nd time we've seen an issue like this. The first time was with Content-MD5 (#675).
This fixes the issue with #780, but there could still be issues in the future. For example, if you take the `s3.getSignedUrl` example in #780, and add the parameter `ContentLanguage`, we'll have the same issue.

Instead, it might be worth taking a look at removing line 74 below. According to the S3 v4 signature docs:
http://docs.aws.amazon.com/AmazonS3/latest/API/sig-v4-header-based-auth.html
`For the purpose of calculating an authorization signature, only the host and any x-amz-* headers are required; however, in order to prevent data tampering, you should consider including all the headers in the signature calculation.`
So it appears we should be able to support signing headers other than `x-amz` headers.

I'm not sure why the `x-amz` check was added below, so we may need to dig into what other ramifications removing that may have.